driver-core/src/main/java/com/datastax/driver/core/DataTypeCqlNameParser.java

```diff
@@ -79,6 +79,9 @@
      */
     static DataType parse(String toParse, Cluster cluster, String currentKeyspaceName, Map<String, UserType> currentUserTypes, Map<String, UserType> oldUserTypes, boolean frozen, boolean shallowUserTypes) {
 
+        if(toParse.startsWith("'"))
```

Comment:
"Custom" types (i.e. types that do not have a CQL representation) appear in schema metadata in the "old" format, i.e., with cassandra internal class names, but surrounded by single quotes.